go/pkg/reader/reader.go

```diff
@@ -182,60 +188,53 @@ func NewCompressedSeeker(fs ReadSeeker) (ReadSeeker, error) {
 		return nil, errors.New("failed creating new encoder")
 	}
 
-	encdW.Encoder.Reset(sb)
+	encdW.Reset(sb)
 	return &compressedSeeker{
 		fs:    fs,
 		encdW: encdW,
 		buf:   sb,
-		err:   nil,
 	}, nil
 }
 
 func (cfs *compressedSeeker) Read(p []byte) (int, error) {
-	if cfs.err != nil {
-		return 0, cfs.err
+	if !cfs.IsInitialized() {
+		return 0, errNotInitialized
 	}
 
-	var n int
-	var errR, errW error
-	for cfs.encdW != nil && cfs.buf.Len() < len(p) && errR == nil && errW == nil {
+	var err error
+	// Repeatedly encode chunks of input data until we have enough compressed
+	// data to fill the output buffer. It can't be known ahead of time how much
+	// uncompressed data will correspond to a certain amount of compressed data,
+	// hence the need for a loop.
+	for cfs.encdW != nil && cfs.buf.Len() < len(p) && err == nil {
```

Comment:
Nit: It would be great if you can reorder the conditionals as:
```suggestion
	for cfs.buf.Len() < len(p) && err == nil && cfs.encdW != nil {
```
That way, the functional conditional is read first, that is to say the loop is for filling up `cfs.buf` by at least `len(p)`.
It would also be great if you added comments to explain the other two conditionals: `err != nil` means the _loop_ hasn't encountered any errors and `encdW != nil` means the _reader_ hasn't encountered any errors (including EOF).

Reply:
Thanks for the suggestion.